src/main/java/teammates/ui/webapi/GetCoursesAction.java

```diff
@@ -47,49 +51,110 @@ public JsonResult execute() {
+                            .collect(Collectors.toList());
+            sqlCourses = sqlLogic.getCoursesForInstructors(sqlInstructors);
+
             break;
         case Const.CourseStatus.ARCHIVED:
             instructors = logic.getInstructorsForGoogleId(userInfo.id)
                     .stream()
                     .filter(InstructorAttributes::isArchived)
                     .collect(Collectors.toList());
             courses = getCourse(instructors);
+
+            sqlInstructors = sqlLogic.getInstructorsForGoogleId(userInfo.id)
+                    .stream()
+                    .filter(Instructor::getIsArchived)
+                    .collect(Collectors.toList());
+            sqlCourses = sqlLogic.getCoursesForInstructors(sqlInstructors);
+
             break;
         case Const.CourseStatus.SOFT_DELETED:
             instructors = logic.getInstructorsForGoogleId(userInfo.id);
             courses = getSoftDeletedCourse(instructors);
+
+            sqlInstructors = sqlLogic.getInstructorsForGoogleId(userInfo.id);
+            sqlCourses = sqlLogic.getSoftDeletedCoursesForInstructors(sqlInstructors);
+
             break;
         default:
             throw new InvalidHttpParameterException("Error: invalid course status");
         }
 
+        courses = courses.stream()
+                .filter(course -> !isCourseMigrated(course.getId()))
+                .collect(Collectors.toList());
+
         Map<String, InstructorAttributes> courseIdToInstructor = new HashMap<>();
         instructors.forEach(instructor -> courseIdToInstructor.put(instructor.getCourseId(), instructor));
 
+        Map<String, Instructor> sqlCourseIdToInstructor = new HashMap<>();
+        sqlInstructors.forEach(instructor -> sqlCourseIdToInstructor.put(instructor.getCourseId(), instructor));
+
         CourseAttributes.sortById(courses);
-        CoursesData coursesData = new CoursesData(courses);
-        coursesData.getCourses().forEach(courseData -> {
-            InstructorAttributes instructor = courseIdToInstructor.get(courseData.getCourseId());
+
+        CoursesLogic.sortById(sqlCourses);
+
+        CoursesData coursesData = new CoursesData(sqlCourses);
+
+        List<CourseData> courseData = coursesData.getCourses();
+
+        List<CourseData> datastoreCourseData =
+                courses.stream().map(CourseData::new).collect(Collectors.toList());
+
+        courseData.addAll(datastoreCourseData);
+
+        // TODO: Remove once migration is completed
+        courseData.sort(Comparator.comparing(CourseData::getCourseId));
+
+        courseData.forEach(cData -> {
+            InstructorAttributes instructor = courseIdToInstructor.get(cData.getCourseId());
             if (instructor == null) {
                 return;
             }
             InstructorPermissionSet privilege = constructInstructorPrivileges(instructor, null);
-            courseData.setPrivileges(privilege);
+            cData.setPrivileges(privilege);
```

Comment:
Let's try to keep the original name for this variable. In general, try to avoid use of abbreviated names in vanilla java for consistency reasons unless there's a clear benefit or abbreviation is widely used (e.g. id). Unlike some of the newer languages out there, it's not really idiomatic to shorten variable names this way + the rest of the file doesn't abbreviate like this. Caught me off guard when reading this.

Reply:
Sorry about it! Had a hard time thinking of the name too to avoid using `*List`. Not sure if this is okay, seems like bringing back the `*List` makes sense for this case..
```java
        CoursesData coursesData = new CoursesData(sqlCourses);

        List<CourseData> coursesDataList = coursesData.getCourses();

        List<CourseData> datastoreCourseData =
                courses.stream().map(CourseData::new).collect(Collectors.toList());

        coursesDataList.addAll(datastoreCourseData);

        // TODO: Remove once migration is completed
        coursesDataList.sort(Comparator.comparing(CourseData::getCourseId));

        coursesDataList.forEach(courseData -> {
            InstructorAttributes instructor = courseIdToInstructor.get(courseData.getCourseId());
            if (instructor == null) {
                return;
            }
            InstructorPermissionSet privilege = constructInstructorPrivileges(instructor, null);
            courseData.setPrivileges(privilege);
        });
```